### ISSUE TITLE
Multiple PHP and HTML homepage templates cause blank Editor

### ISSUE THREAD

**Issue by mikachan** [WordPress/gutenberg#64303]
### Description

If there are multiple home templates (index, home, front-page) in both PHP and HTML formats, the Editor doesn't seem to understand what to render and will render nothing, with no console errors:

![image](https://github.com/user-attachments/assets/460163f2-99bb-4274-a4a0-396cfc4d9fe6)

I would expect the Editor to either default to the nearest HTML template or show a helpful warning/error message.

### Step-by-step reproduction instructions

1. Add a home.php, front-page.php, index.php and index.html templates to the active theme
2. Attempt to load the Editor at wp-admin/site-editor.php
3. See a blank Editor screen

### Environment info

- WordPress 6.6.1, GB trunk
- Seen while testing the Humanity theme: https://github.com/amnestywebsite/humanity-theme

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by carolinan** [comment]
Is this related to https://github.com/WordPress/gutenberg/issues/63124 ?
But the Site Preview is not black for me but white with the loader showing.

**Comment by carolinan** [comment]
If there is a PHP equivalent, which is showing on the front, I don't think that the HTML template should be showing in the preview, since the editor and front do not match.

There is also the scenario where there is only a PHP template, and no HTML template to fall back to.

**Comment by mikachan** [comment]
> Is this related to https://github.com/WordPress/gutenberg/issues/63124 ?

Yes I think so! Happy to close this as a duplicate. I was looking for existing issues using "blank editor", "editor broken" 😅 Although I guess we're seeing slightly different things, as you have the white background with the loader, and in my case I saw only the grey background colour. But I'm pretty confident it's the same underlying issue.

> If there is a PHP equivalent, which is showing on the front, I don't think that the HTML template should be showing in the preview, since the editor and front do not match.

I agree, that makes sense. We should have a way of communicating this in the Editor, maybe with a `<Warning>` as you said [here](https://github.com/WordPress/gutenberg/issues/63124#issuecomment-2255426394).

> There is also the scenario where there is only a PHP template, and no HTML template to fall back to.

Yes, these templates don't show in the Editor at all, which suggests that they don't exist. Again it would be ideal to communicate this somehow in the Editor, even if it's a warning message.

**Comment by hanneslsm** [comment]
A recent issue with a black editor: https://github.com/WordPress/gutenberg/issues/64220
@Mamaduka (since you have tested and self-assigned the other issue already)

**Comment by mikachan** [comment]
> A recent issue with a black editor: https://github.com/WordPress/gutenberg/issues/64220

I believe this issue is caused by a different error, although the result is certainly very similar, with the blank editor.

**Comment by youknowriad** [comment]
Closing this as a duplicate per @mikachan's comment above — the canonical issue for hybrid-theme template resolution showing the wrong/blank editor is [#64341](https://github.com/WordPress/gutenberg/issues/64341) (this was first merged into #63124, which was then consolidated there too). Please follow along and add any details over there!
